README.md

```diff
@@ -67,38 +73,64 @@ plugins:
 custom:
   vpcDiscovery:
     vpcName: '${opt:env}'
-    subnetNames: # optional if securityGroupNames are specified
-      - '${opt:env}_<name of subnet>'
-    securityGroupNames: # optional if subnetNames are specified
-      - '${opt:env}_<name of security group>'
+    # optional if `securityGroups` option is specified
+    # list of tag key and values 
+    subnets:
+      # tag key
+      - tagKey: <tag_name>
+        # an array of values
+        tagValues:
+          - '<tag_value>'
+    # optional if `subnets` option is specified
+    # list of tag key and value or names
+    securityGroups:
+      # tag key
+      - tagKey: <tag_name>
+        # an array of values
+        tagValues:
+          - '<tag_value>'
+      # optional if `tagKey` and `tagValues` are specified
+      - names:
+        - '<security_group_name>'
 
-# (Optional) set a config for the specific function
+# Optional: Either set `custom.vpcDiscovery` or `functions.<function name>.vpcDiscovery`
 functions:
-  example1:
+  example:
     handler: handler.example
-    # inherit `custom.vpcDiscovery` config if specified
+    # inherit `custom.vpcDiscovery` config in case `custom.vpcDiscovery` is specified
   example2:
     handler: handler.example
-    # skip vpc configuration
+    # skip vpc configuration for the current function
     vpcDiscovery: false
   example3:
     handler: handler.example
-    # inherit `custom.vpcDiscovery` and override security group names
+    # inherit `custom.vpcDiscovery` config in case `custom.vpcDiscovery` is specified and override security group names
     vpcDiscovery:
       vpcName: '${opt:env}'
-      securityGroupNames:
-        - '${opt:env}_<name of security group>'
+      securityGroups:
+        - tagKey: <tag_name>
+          tagValues:
+            - '<tag_value>'
+        ...
   example4:
     handler: handler.example
-    # override basic subnet names and security group names
+    # override or set basic subnets and security groups items
     vpcDiscovery:
       vpcName: '${opt:env}'
-      subnetNames: # optional if securityGroupNames are specified
-        - '${opt:env}_<name of subnet>'
-      securityGroupNames:  # optional if subnetNames are specified
-        - '${opt:env}_<name of security group>'        
+      subnets: # optional if `subnets` option is specified
+        - tagKey: <tag_name>
+          tagValues:
+            - '<tag_value>'
+        ...
+      securityGroups: # optional if `subnets` option is specified
+        - tagKey: <tag_name> # optional if `names` option is specified
+          tagValues: # optional if `names` option is specified
+            - '<tag_value>'
+        - names: # optional if `tagKey` and `tagValues` are specified
+          - '${opt:env}_<name of security group>'
```

Comment:
This might be simpler. We can leave it up to users to namespace security groups by environment or just use tags instead.
```suggestion
          - '<name of security group>'
```